cartridge/utils.lua

```diff
@@ -333,6 +339,41 @@ local function wait_lsn(id, lsn, pause, timeout)
     end
 end
 
+
+-- Set FD_CLOEXEC flag for the given file descriptor.
+--
+-- See: https://www.gnu.org/software/libc/manual/html_node/Descriptor-Flags.html
+--
+-- @function
+-- @local
+--
+-- @tparam number fd
+--
+-- @treturn[1] boolean true
+-- @treturn[2] nil
+-- @treturn[2] table Error description
+local function fd_cloexec(fd)
+    checks('number')
+
+    local ret = ffi.C.fcntl(fd, F_GETFD)
+    if ret < 0 then
+        return nil, FcntlError:new(
+            "fcntl(F_GETFD) failed: %s", errno.strerror()
+        )
+    end
+
+    local flags = ffi.cast('uint64_t', bit.bor(ret, FD_CLOEXEC))
```

Comment:
I think you could check if such flag is already set. And if `bit.band(ret, FD_CLOEXEC) ~= 0` you can return true immediately.

Reply:
Agree, fixed.